### DISCUSSION TITLE
Sharing form values in funnel wizard

### DISCUSSION THREAD

**Discussion by alifa20** [react-hook-form/react-hook-form#8093] [Q&A] [answered]
🐨  I am using: Ts, zod, useReducer, and rhf useForm
🚨  I am not using react router.
Take following case as an example: 
Form --> Review --> api (might give an error) --> Form (with an error on specific field)
Once the user filled the Form, they will be presented with Review screen which reads the form values and shows them to the user for confirmation. 
My form has validation on `age` field to be a minimum of 18 but the api might give back an error stating the minimum age should be 21.
The api returns an error and we need to navigate the user to that initial form page and highlight the age input with an error that we got from the api.

I have 2 solutions for this case and wanted to get ideas as to which one is a better solution
1. Pull out the useForm hook into the parent component (index.tsx) so that both Form.tsx and Review.tsx files have access to the form values.
https://codesandbox.io/s/funnel-parent-hook-ugykg3

2. Dispatch the values and grab a copy of form values into the useReducer's state. 
https://codesandbox.io/s/funnel-child-v6jmrd?file=/src/Form.tsx

#1 is good because: 
- I can rely on rhf to make sure the values/isDirty and so on are in sync when the user goes back and forth between these screens.
- I can get access to setError and other methods of useForm in my child components (ie. Review.tsx)

#2 I guess is more aligned with what is mentioned: https://react-hook-form.com/advanced-usage#WizardFormFunnel

🚨  Please note, in my example, I don't plan to use react-router and don't plan to change the page address as the user navigates between form and review

Question 1:  which one is a better way to solve this?
Question 2: in #2 (grab a copy of form data and save in useReducer) How can I make the error appear when the user lands back in the Form?

💥 I highly appreciate your time and sharing your thoughts

**Comment by bluebill1049** [comment, marked as answer]
Hello 👋🤩 I would go with break each step as an individual form and have a state management library host all the submitted and valid data.

**Reply by alifa20** [reply]
Thanks for the prompt reply mate 😄 🙏

**Reply by bluebill1049** [reply]
you are always welcome mate. I was in a meeting, let me give a bit more context and reasoning.

Props

- better standard for form submission (close to native behavior)
- clearer contract/schema for each part of your applications
- easy to debug and manages

Cons

- more code
- potentially more schemas depend on how you manage your schema

In general, I prefer separate concerns, it seems to be easier to manage when your application forms grow.

**Reply by alifa20** [reply]
Yeah thanks for the reply. The `more code` is a bit that tempted me a bit. But yeah I get your points 👍
